src/covid19sim/base.py

```diff
@@ -37,16 +37,7 @@ def timestamp(self):
         Returns:
             datetime.datetime: Current date.
         """
-        #
-        ##
-        ## The following is preferable, but not equivalent to the initial
-        ## version, because timedelta ignores Daylight Saving Time.
-        ##
-        #
-        #return datetime.datetime.fromtimestamp(int(self.now))
-        #
-        return self.initial_timestamp + datetime.timedelta(
-            seconds=self.now-self.ts_initial)
+        return datetime.datetime.fromtimestamp(int(self.now))
```

Comment:
Was this the most time-consuming part of the code? Is it being solved through this change?

Reply:
This is not the most time-consuming part of the code, far from it. `datetime+timedelta` *might* even be faster than this new line of code. I am addressing one problem here, while setting the stage for solving another.

1. `datetimetime+timedelta` *can* generate either nonsense timestamps ("spring forward": For instance, March 8, 02:00-03:00 never existed, and the default simulation configuration happens to start Feb. 28 and runs over this switch) or ambiguous dates ("fall back", which will next happen when the clocks are wound back one hour on November 1, 02:00).

  ```
>>> # Spring Forward
>>> datetime.datetime.fromtimestamp(1583650799)
datetime.datetime(2020, 3, 8, 1, 59, 59)
>>> datetime.datetime.fromtimestamp(1583650800)
datetime.datetime(2020, 3, 8, 3, 0)
>>> 
>>> # Fall Back
>>> datetime.datetime.fromtimestamp(1604206800)
datetime.datetime(2020, 11, 1, 1, 0)
>>> datetime.datetime.fromtimestamp(1604210399)
datetime.datetime(2020, 11, 1, 1, 59, 59)
>>> datetime.datetime.fromtimestamp(1604210400)
datetime.datetime(2020, 11, 1, 1, 0, fold=1)
>>> datetime.datetime.fromtimestamp(1604213999)
datetime.datetime(2020, 11, 1, 1, 59, 59, fold=1)
>>> datetime.datetime.fromtimestamp(1604214000)
datetime.datetime(2020, 11, 1, 2, 0)
  ```
  Clocks are therefore not *monotonic*, nor unambiguous. But simulator time _must be_; For that reason I have chosen POSIX time (monotonic, never ambiguous, not affected by DST).

2. The virus does not care about DST. Time-exposure to the virus does not depend on DST. By standardizing on integer number of seconds from the Epoch, arithmetic on timestamps becomes trivial integer comparison/subtraction/addition, and not `datetime` operations. Operations on time involving temporal overlaps, viral load, infectiousness and logging are extremely common and do not require `datetime` representations. This is where we have potential for optimization.

3. Only a few cases exist where knowing the hour of the day (adjusted for DST) is important, because our `Humans` are programmed with specific habits and they are expressed as _hours of the day_. Computing decisions about when to go shopping/sleeping/... will thus require either `datetime.datetime.fromtimestamp(now)` or `time.localtime(now)`. It is important to highlight here that humans only perform so many shopping/gym visits per week (requiring DST awareness), whereas the viral dynamics computations (which ignore DST) are performed hundreds or thousands of times on every human a day.

4. For logging purposes, "pretty-printed" human-readable time (with timezone/DST applied) is ultimately a *presentation* problem. POSIX timestamps are agnostic to how they are printed; `datetime` implicitly carries a judgement about both timezone and DST.

5. For data output purposes, both `datetime` and `int` are picklable, but `int` is much smaller and is a primitive type.

Reply:
It is my intent, in a future series based atop this PR, to change the type of all `*_timestamp` attributes from `datetime` to `int`, thereby reducing and potentially eliminating usage of `env.timestamp`. That will be an invasive change. For the purpose of keeping something approximating backwards compatibility in this PR, I am not deleting that attribute yet.

Reply:
Thanks for the detailed explanation!